src/library/scala/util/matching/Regex.scala

```diff
@@ -341,7 +341,7 @@ class Regex private[matching](val pattern: Pattern, groupNames: String*) extends
   def unapplySeq(target: Any): Option[List[String]] = target match {
     case s: CharSequence =>
       val m = pattern matcher s
-      if (runMatcher(m)) Some((1 to m.groupCount).toList map m.group)
+      if (runMatcher(m)) Some((1 to m.groupCount).map(m.group(_))(collection.breakOut))
```

Comment:
The following is a bit more direct and even avoids allocation of the `Range`.

```scala
List.tabulate(m.groupCount)(i => m.group(i + 1))
```

Reply:
Thank you. I will try to make the change and get some profiling data

Reply:
Hi @retronym, it cost me few days to get familiar with jmh, could you please review my PR and below benchmark result?

2.12.x is my baseline (I am using a0026c5, because if I merged latest commit, my local build would fail during build compilation)

cong/2.12.x_Regex is my previous fix like:
`Some((1 to m.groupCount).map(m.group(_))(collection.breakOut))`

Jason is the benchmark for current PR (I adopted your suggestion: 
`List.tabulate(m.groupCount)(i => m.group(i + 1))`